pkg/operator/ceph/disruption/nodedrain/reconcile.go

```diff
@@ -127,19 +127,32 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 			}
 		}
 	}
+
+	controllerBool := false
+	blockOwnerDeletionBool := false
 	// Create or Update the deployment default/foo
 	deploy := &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      k8sutil.TruncateNodeName(fmt.Sprintf("%s-%%s", CanaryAppName), nodeHostnameLabel),
 			Namespace: r.context.OperatorNamespace,
+			OwnerReferences: []metav1.OwnerReference{
```

Comment:
@leseb I am not using `SetControllerReference` because I want to set the OwnerReference.Controller: false

Reply:
Why use the node as the owner reference? What seems more natural is to add an owner reference to all the CephCluster instances. Then as soon as all the CephCluster CRs are deleted, there is no need to manage the PDBs and the canaries can be removed, right? In the case of a node getting removed, the operator can watch that case and remove the canary, right? The operator will also be monitoring to see if all ceph daemons have been removed from a node and remove the canary from that node as well, right?

Reply:
It still seems odd to have the node as the owner reference rather than a rook component. If rook creates it, rook should own it. What about setting the owner reference to the rook-ceph-operator deployment? Then whenever the operator is deleted, the canaries will be cleaned up. But if the operator is still running, it will manage creation/deletion of the canaries when nodes are added/removed or the ceph cluster is configured.

Reply:
That works, I'll add that!